### PR TITLE
Use device ML service for captions.

### PR DESCRIPTION
**Pull request by yuangu-google**

### CODE REVIEW

File: shells/pipes-shell-2/web/deploy/source/index.html [hunk 1]
```diff
@@ -11,26 +11,46 @@
...
 
   <meta charset="utf-8">
   <meta name="viewport" content="width=device-width, user-scalable=no, minimum-scale=1.0, maximum-scale=1.0">
 
-  <script type="module" src="./shell.js"></script>
-
   <style>
-    body, input, button {
-      font-family: sans-serif;
+    body {
```
Comment (shaper) [inline shells/pipes-shell-2/web/deploy/source/index.html]: Move to a separate PR.

File: shells/pipes-shell-2/bus.js [hunk 2]
```diff
@@ -8,6 +8,7 @@
  * http://polymer.github.io/PATENTS.txt
  */
 
+import {Services} from '../../build/runtime/services.js';
```
Comment (shaper) [inline shells/pipes-shell-2/bus.js]: Here we should wait to make sure @sjmiles is ok with the changes. Can you send him a pointer to the other internal change so he can see the other side of it?

File: particles/PipeApps2/source/CaptionThis.js [hunk 3]
```diff
@@ -16,14 +16,29 @@ defineParticle(({DomParticle, html, log}) => {
...
+    async classify(text) {
+      this.startBusy();
+      const rawResponse =
+          await this.service({call: 'textclassifier.classifyText', text});
+      const response = JSON.parse(rawResponse);
+      const json = JSON.stringify(
+          response && response.results &&
+          response.results
+          .map(
+              x => (x && x.entities && x.entities.length
```
Comment (shaper) [inline particles/PipeApps2/source/CaptionThis.js]: Add a comment to document what TC is returning here.

File: shells/pipes-shell-2/bus.js (hunk 2 above)
Comment (sjmiles) [reply shells/pipes-shell-2/bus.js]: The `bus` itself is generic, so this is not in the right place. 

The intent of the pipe-shell-2 upgrade was to force all correspondence between the two sides to happen via messages,  binding more code directly from Java is a different architecture.

Maybe this was the easiest way to get from point A to point B. We can discuss, but at the very least this shouldn't be in `bus.js`.

File: shells/pipes-shell-2/web/deploy/source/index.html [hunk 4]
```diff
@@ -11,26 +11,46 @@
...
+      min-width: 320px;
+      overflow-x: hidden;
+      overflow-y: scroll;
+      background-color: white;
     }
-    [slotid="root"] {
-      padding: 4px;
-      border: 1px dotted gray;
+    body, button, input {
+      font-family: 'Google Sans', sans-serif;
```
Comment (sjmiles) [inline shells/pipes-shell-2/web/deploy/source/index.html]: It's my expectation that 'sans-serif' will call up the correct Sans for the platform (used to be Roboto, is it Google Sans now?) ... what did you see?

File: shells/pipes-shell-2/web/deploy/source/index.html (hunk 1 above)
Comment (sjmiles) [reply shells/pipes-shell-2/web/deploy/source/index.html]: Maybe I'm missing something Walt, but this seems ok to me. IMO the point of the `index.html` is high-level configuration, and it seems like all of these changes (styling too) are aimed at tailoring the app for the Android platform, which strikes me as proper.

File: shells/pipes-shell-2/web/deploy/source/index.html [hunk 5]
```diff
@@ -11,26 +11,46 @@
...
+        xhr.onerror = function() {
+          reject(new TypeError('Local request failed for ' + url))
+        }
+        xhr.open('GET', url)
+        xhr.send(null)
+      })
+    };
+  </script>
+
+  <script src="./paths.js"></script>
```
Comment (sjmiles) [inline shells/pipes-shell-2/web/deploy/source/index.html]: `paths.js` should be baked into `shell.js` already

File: shells/pipes-shell-2/web/deploy/source/index.html (hunk 4 above)
Comment (yuangu-google) [reply shells/pipes-shell-2/web/deploy/source/index.html]: This page is not shown so I think the font doesn't matter. I removed Google Sans as sans-serif looks more universal too me.

**Comment by yuangu-google** [comment]
PTAL, thanks.

File: particles/PipeApps2/source/CaptionThis.js [hunk 6]
```diff
@@ -16,14 +16,30 @@ defineParticle(({DomParticle, html, log}) => {
     get template() {
       return `<span>{{json}}</span>`;
     }
-    render({query}) {
```
Comment (sjmiles) [inline particles/PipeApps2/source/CaptionThis.js]: Fwiw, using `render` instead of `update` is a trick to avoid SpecEx running expensive operations (like ML inferencing) when using Planner.

We aren't using Planner atm, so this change is fine, just want you to have the info.

File: particles/PipeApps2/source/CaptionThis.js [hunk 7]
```diff
@@ -16,14 +16,30 @@ defineParticle(({DomParticle, html, log}) => {
...
-        this.updateSingleton('output', {json});
-        return {json};
+        this.classify(query.name);
       }
     }
-  };
 
+    async classify(text) {
+      this.startBusy();
+      const rawResponse =
```
Comment (sjmiles) [inline particles/PipeApps2/source/CaptionThis.js]: In general, we should use try/catch or try/finally to make sure 'doneBusy` is called when there is an exception or the system can get locked up.
